pkg/cli/image/manifest/security.go

```diff
@@ -112,7 +115,60 @@ func (o *SecurityOptions) NewContext() (*registryclient.Context, error) {
 		}
 	}
 
-	context := registryclient.NewContext(rt, insecureRT).WithCredentials(creds).WithICSP(o.ICSPFile, o.LookupClusterICSP, o.ICSPClientFn)
-	context.DisableDigestVerification = o.SkipVerification
-	return context, nil
+	var regContext *registryclient.Context
+	imageSources := imagecontentsources.ICSPSources{
+		ICSPFile:     o.ICSPFile,
+		ICSPClientFn: o.ICSPClientFn,
+	}
+	regContext = registryclient.NewContext(rt, insecureRT).WithCredentials(creds)
```

Comment:
yeah, so with my proposed changes from https://github.com/openshift/library-go/pull/939 this would be as simple as:
```golang
regContext = registryclient.NewContext(rt, insecureRT).WithCredentials(creds)
if len(o.ICSPFile) > 0 || o.LookupClusterICSP {
    regContext.WithImageResolver(imagecontentsources.NewICSPSources(o.ICSPFile, o.ICSPClientFn)
}
```

Reply:
updated, got it